### PR TITLE
KEYCLOAK-15477 Fix permission evaluation logic

### PR DESCRIPTION
**Pull request by hmlnarik**

### CODE REVIEW

File: services/src/main/java/org/keycloak/services/resources/admin/AdminConsole.java [hunk 1]
```diff
@@ -232,28 +233,36 @@ public Response whoAmI(final @Context HttpHeaders headers) {
...
-            if (!realmAdminAccess.containsKey(realm.getName())) {
-                realmAdminAccess.put(realm.getName(), roles);
-            } else {
-                realmAdminAccess.get(realm.getName()).addAll(roles);
-            }
+    private static <T> HashSet<T> union(Set<T> set1, Set<T> set2) {
+        if (set1 == null && set2 == null) {
+            return null;
+        }
+        HashSet<T> res = (set1 instanceof HashSet) ? (HashSet <T>) set1 : new HashSet<>(set1);
```
Comment (sguilhen) [inline services/src/main/java/org/keycloak/services/resources/admin/AdminConsole.java:252]: Isn't there a NPE waiting to happen here if set1 is null? IIRC the HasSet constructor raises an NPE if the collection specified in the constructor is null.

File: testsuite/integration-arquillian/tests/base/src/test/java/org/keycloak/testsuite/admin/AdminConsolePermissionsCalculatedTest.java [hunk 2]
```diff
@@ -0,0 +1,65 @@
+package org.keycloak.testsuite.admin;
```
Comment (sguilhen) [inline testsuite/integration-arquillian/tests/base/src/test/java/org/keycloak/testsuite/admin/AdminConsolePermissionsCalculatedTest.java:1]: License header

**Review by sguilhen — CHANGES_REQUESTED** [review]

File: services/src/main/java/org/keycloak/services/resources/admin/AdminConsole.java (hunk 1 above)
Comment (hmlnarik) [reply services/src/main/java/org/keycloak/services/resources/admin/AdminConsole.java]: Good catch, indeed!

Comment (hmlnarik) [reply services/src/main/java/org/keycloak/services/resources/admin/AdminConsole.java]: Good catch, indeed!

**Comment by hmlnarik** [comment]
Thank you for the review! Review comments should be fixed by now.

**Review by sguilhen — APPROVED** [review]

**Review by vmuzikar — APPROVED** [review]
LGTM, thank you @hmlnarik!

**Review by vramik — APPROVED** [review]

**Review by mposolda — APPROVED** [review]

**Pull request merged**